pkg/query-service/app/traces/v3/query_builder.go

```diff
@@ -501,19 +501,19 @@ func addOffsetToQuery(query string, offset uint64) string {
 // step is in seconds
 func PrepareTracesQuery(start, end int64, panelType v3.PanelType, mq *v3.BuilderQuery, options v3.QBOptions) (string, error) {
 	// adjust the start and end time to the step interval
-	start = start - (start % (mq.StepInterval * 1000))
-	end = end - (end % (mq.StepInterval * 1000))
+	startForGraphQuery := start - (start % (mq.StepInterval * 1000))
```

Comment:
The change is only handling the graph with limits. The graph requests without limits are not handled. Alternatively, you could make this step skip for panel type List/Trace.